sway-core/src/ir_generation/function.rs

```diff
@@ -128,40 +128,40 @@ impl<'te> FnCompiler<'te> {
         &mut self,
         context: &mut Context,
         md_mgr: &mut MetadataManager,
-        ast_node: ty::TyAstNode,
+        ast_node: &ty::TyAstNode,
     ) -> Result<Option<Value>, CompileError> {
         let span_md_idx = md_mgr.span_to_md(context, &ast_node.span);
-        match ast_node.content {
+        match &ast_node.content {
             ty::TyAstNodeContent::Declaration(td) => match td {
                 ty::TyDeclaration::VariableDeclaration(tvd) => {
-                    self.compile_var_decl(context, md_mgr, *tvd, span_md_idx)
+                    self.compile_var_decl(context, md_mgr, tvd, span_md_idx)
                 }
                 ty::TyDeclaration::ConstantDeclaration(decl_id) => {
-                    let tcd = declaration_engine::de_get_constant(decl_id, &ast_node.span)?;
+                    let tcd = declaration_engine::de_get_constant(decl_id.clone(), &ast_node.span)?;
                     self.compile_const_decl(context, md_mgr, tcd, span_md_idx)?;
                     Ok(None)
                 }
                 ty::TyDeclaration::FunctionDeclaration(_) => {
                     Err(CompileError::UnexpectedDeclaration {
                         decl_type: "function",
-                        span: ast_node.span,
+                        span: ast_node.span.clone(),
```

Comment:
Also while returning error's I am `clone`ing spans. I feel like this should be ok because of the same reason with previous comment but open to suggestions again 🙌

Reply:
Cloning for errors is fine IMO. :+1: